### PR TITLE
Added related articles.

### PR DESCRIPTION
**Pull request by tan-90**
Closes WEB-60

### CODE REVIEW

File: src/components/RelatedArticles/RelatedArticles.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  article: PostProps
+  count?: number
+}
+export default function RelatedArticles({ article, count = 3 }: RelatedArticlesProps) {
+  return (
+    <StaticQuery<GatsbyTypes.RelatedArticlesQuery>
+      query={relatedArticlesQuery}
+      render={(data) => {
+        const allArticles = data.allMarkdownRemark.edges.map(({ node }) => node)
+        return <PostGrid posts={getRelatedArticles(article, allArticles, count)} name='Related Articles' narrow />
```
Comment (Lisenish) [inline src/components/RelatedArticles/RelatedArticles.tsx:17]: So if we don't have any related articles there will be only title? Maybe we shouldn't show related articles block at all then?

**Review by Lisenish — CHANGES_REQUESTED** [review]
Only minor things, mostly questions

File: src/components/RelatedArticles/RelatedArticles.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+import { mapToPost, PostProps } from '../Post/Post'
+import PostGrid from '../PostGrid/PostGrid'
+
+export interface RelatedArticlesProps {
+  article: PostProps
+  count?: number
+}
+export default function RelatedArticles({ article, count = 3 }: RelatedArticlesProps) {
+  return (
+    <StaticQuery<GatsbyTypes.RelatedArticlesQuery>
```
[the remark this answers is not in the document]
Comment (tan-90) [reply src/components/RelatedArticles/RelatedArticles.tsx:13]: I tried the hook, but it breaks the entire preview when it fails to fetch, the render props version breaks more nicely. The fetch should never fail on production, but I just thought it was better to keep the safer version. If needed, it's an easy change though.

File: src/components/RelatedArticles/RelatedArticles.tsx (hunk 1 above)
Comment (tan-90) [reply src/components/RelatedArticles/RelatedArticles.tsx:17]: Oh, that was an oversight, my bad. Fixing now.

File: src/components/PostGrid/PostGrid.tsx [hunk 3]
```diff
@@ -10,8 +10,9 @@ export interface PostGridProps {
   name?: string
   link?: React.ReactNode
   tags?: string[]
+  narrow?: boolean
```
[the remark this answers is not in the document]
Comment (tan-90) [reply src/components/PostGrid/PostGrid.tsx:13]: Went with "perRow" here, default being four and narrow being three.

File: src/components/RelatedArticles/RelatedArticles.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+  query RelatedArticles {
+    allMarkdownRemark(
+      filter: {
+        fileAbsolutePath: { regex: "/(blog)/.*\\.md$/" }
+      }
+      sort: { order: DESC, fields: frontmatter___publishDate }
+      limit: 1000
+    ) {
+      edges {
+        node {
```
[the remark this answers is not in the document]
Comment (tan-90) [reply src/components/RelatedArticles/RelatedArticles.tsx:33]: Good point, I forgot that they worked across files.

File: src/components/RelatedArticles/RelatedArticles.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (Lisenish) [reply src/components/RelatedArticles/RelatedArticles.tsx:13]: Ok no, no need to change it if it's useful I just thought maybe it's some overlook.

**Review by Lisenish — APPROVED** [review]

**Pull request merged**